app/services/open_investment_time_calculator.rb

```diff
@@ -0,0 +1,68 @@
+class OpenInvestmentTimeCalculator
+  def initialize(time_entries)
+    @newest_time_entries = time_entries
+    @aggregated_times_per_employee = nil
+
+    @employee_credentials = nil
+    @employee_id = nil
+    @employee_time_entries = nil
+    @employee = nil
+    @employee_investment_time = nil
+
+    @ratio_of_investment_time = 4
+  end
+
+  def add_time_entries_to_employees
+    @aggregated_times_per_employee = @newest_time_entries.map do |time_entries_per_employee|
+      extract_information(time_entries_per_employee)
+      add_entries_to_investment_time
+      set_new_investment_time
+      @employee
```

Comment:
I consider this to be dangerous. Are you sure that the `@aggregated_times_per_employee`-*Enumerable* will not only be full of references to the last object you worked with?